src/rdd/rdd.rs

```diff
@@ -695,6 +695,59 @@ pub trait Rdd: RddBase + 'static {
             second,
         ))
     }
+
+    fn intersection<T>(
```

Comment:
we probably want to name this `intersection_with_num_partitions` since we don't have method overloading in Rust, following what we have done in other methods (e.g. distinct).

Reply:
> we probably want to name this `intersection_with_num_partitions` since we don't have method overloading in Rust, following what we have done in other methods (e.g. distinct).

Ok. And do we need a method like `intersection_with_partitioner` that it exists in implement of Apache Spark

Reply:
Probably, but you can do that in an other pr if you wish.

Reply:
agreeing with @iduartgomez . Shall I leave this PR open for you to add them or merge it now so that you can add it later in another PR?

Reply:
> agreeing with @iduartgomez . Shall I leave this PR open for you to add them or merge it now so that you can add it later in another PR?

leave it after I add them is ok.